File: mixin-base/src/main/java/net/yakclient/mixin/base/internal/bytecode/MixinCV.java
package net.yakclient.mixin.base.internal.bytecode;

import org.objectweb.asm.ClassVisitor;

import java.util.Map;
import java.util.Queue;

public class MixinCV extends ClassVisitor {
    protected final Map<String, Queue<QualifiedInstruction>> injectors;

    public MixinCV(ClassVisitor cv, Map<String, Queue<QualifiedInstruction>> injectors) {
        super(ByteCodeUtils.ASM_VERSION, cv);
        this.injectors = injectors;
    }

    protected boolean hasInjection(String name) {
        for (var s : this.injectors.keySet()) {
            if (ByteCodeUtils.descriptorsSame(name, s)) return true;
        }
        return false;
    }

    protected Queue<QualifiedInstruction> getInjection(String name) {
        for (var s : this.injectors.keySet()) {
            if (ByteCodeUtils.descriptorsSame(name, s)) return this.injectors.get(s);
        }
        throw new IllegalArgumentException("Failed to find injection");
    }
}

File: base/src/main/kotlin/net/yakclient/mixins/base/internal/bytecode/MixinCV.kt
package net.yakclient.mixins.base.internal.bytecode

import org.objectweb.asm.ClassVisitor
import java.util.*

open class MixinCV(cv: ClassVisitor?, private val injectors: Map<String, Queue<QualifiedInstruction>>) :
    ClassVisitor(ByteCodeUtils.ASM_VERSION, cv) {
    protected fun hasInjection(name: String?): Boolean {
        for (s in injectors.keys) {
            if (ByteCodeUtils.descriptorsSame(name, s)) return true
        }
        return false
    }

    protected fun getInjection(name: String?): Queue<QualifiedInstruction> {
        for (s in injectors.keys) {
            if (ByteCodeUtils.descriptorsSame(name, s)) return injectors[s]!!
        }
        throw IllegalArgumentException("Failed to find injection")
    }
}